### PR TITLE
Make sure the "all" alias is defined in directories without dune files

### LINKED ISSUE #2927: generate_opam_files seems broken in 2.0
I have not been able to generate opam files from dune files in 2.0. Here is a minimal example that works to generate an opam file with dune `1.11.4` but does not lead to a generated opam file for dune `2.0`:

```scheme
(lang dune 1.11)
(name foo)
(homepage https://my.home.page)

(generate_opam_files true)

(package
 (name foo)
 )
```

I'm guessing we also need tests to guard against this regression in the future.

### PR DESCRIPTION
**Pull request by ghost**
Fixes #2927

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -14,6 +14,9 @@
 - Fix `dune --version`. The bootstrap didn't compute the version
   correctly. (#2929, fixes #2911, @diml)
 
+- Make sure the `@all` alias is defined when no `dune` file is present
+  in a directory (#...., fix #2927, @diml)
```
Comment (rgrinberg) [inline CHANGES.md]: 2946

**Review by rgrinberg — COMMENTED** [review]
The JS tests need to be updated.

I'm wondering, is this something that should apply to other aliases as well:
* install
* runtest
* check

Seems like many aliases are better behaved if defined even without a dune file.

**Comment by diml** [comment]
I believe that's already the case. For instance, in the case mentioned in the original report (octavius), `dune build @check` is regenerating the opam file.

**Comment by imbsky** [comment]
What is the status of this PR?

**Comment by rgrinberg** [comment]
Thanks for reminding me. I've rebased and it should be good to go after the CI finishes.

File: test/blackbox-tests/test-cases/dune-project-meta/github2927/run.t [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  > (lang dune 2.0)
+  > (generate_opam_files true)
+  > (package (name foo))
+  > EOF
+
+  $ cd attached-to-all && dune build
+
+  $ cat attached-to-all/foo.opam
+  cat: attached-to-all/foo.opam: No such file or directory
+  [1]
```
Comment (kit-ty-kate) [inline test/blackbox-tests/test-cases/dune-project-meta/github2927/run.t]: Does this really fix the issue? I feel like the opam file should be generated here, not missing

Comment (rgrinberg) [reply test/blackbox-tests/test-cases/dune-project-meta/github2927/run.t]: Good catch. It was a bug in the test itself.

**Pull request merged**